tests/contrib/pytest/test_pytest.py

```diff
@@ -325,6 +325,48 @@ def test_xfail_conditional():
         assert spans[1].get_tag(test.RESULT) == test.Status.XFAIL.value
         assert spans[1].get_tag(test.XFAIL_REASON) == "test should xfail"
 
+    def test_xfail_runxfail_fails(self):
+        """Test xfail with --runxfail flags should not crash when failing."""
+        py_file = self.testdir.makepyfile(
+            """
+            import pytest
+
+            @pytest.mark.xfail(reason='should fail')
+            def test_should_fail():
+                assert 0
+
+        """
+        )
+        file_name = os.path.basename(py_file.strpath)
+        self.inline_run("--ddtrace", "--runxfail", file_name)
```

Comment:
this `--runxfail` flag makes it so that the marked as `xfail` tests run anyway, so no `wasxfail` value is added to the result (https://docs.pytest.org/en/6.2.x/skipping.html#ignoring-xfail)